testing/fpgad/test_command_line_c.cpp

```diff
@@ -0,0 +1,237 @@
+// Copyright(c) 2019, Intel Corporation
+//
+// Redistribution  and  use  in source  and  binary  forms,  with  or  without
+// modification, are permitted provided that the following conditions are met:
+//
+// * Redistributions of  source code  must retain the  above copyright notice,
+//   this list of conditions and the following disclaimer.
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution.
+// * Neither the name  of Intel Corporation  nor the names of its contributors
+//   may be used to  endorse or promote  products derived  from this  software
+//   without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING,  BUT NOT LIMITED TO,  THE
+// IMPLIED WARRANTIES OF  MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED.  IN NO EVENT  SHALL THE COPYRIGHT OWNER  OR CONTRIBUTORS BE
+// LIABLE  FOR  ANY  DIRECT,  INDIRECT,  INCIDENTAL,  SPECIAL,  EXEMPLARY,  OR
+// CONSEQUENTIAL  DAMAGES  (INCLUDING,  BUT  NOT LIMITED  TO,  PROCUREMENT  OF
+// SUBSTITUTE GOODS OR SERVICES;  LOSS OF USE,  DATA, OR PROFITS;  OR BUSINESS
+// INTERRUPTION)  HOWEVER CAUSED  AND ON ANY THEORY  OF LIABILITY,  WHETHER IN
+// CONTRACT,  STRICT LIABILITY,  OR TORT  (INCLUDING NEGLIGENCE  OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,  EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+#include "test_system.h"
+
+extern "C" {
+
+#include "fpgad/api/logging.h"
+#include "fpgad/command_line.h"
+
+bool cmd_register_null_gbs(struct fpgad_config *c, char *null_gbs_path);
+
+}
+
+#include <config.h>
+#include <opae/fpga.h>
+
+#include <fstream>
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
+#include <errno.h>
+#include <unistd.h>
+#include <linux/limits.h>
+#include <sys/types.h>
+#include <sys/wait.h>
+#include "gtest/gtest.h"
+
+using namespace opae::testing;
+
+class fpgad_command_line_c_p : public ::testing::TestWithParam<std::string> {
+ protected:
+  fpgad_command_line_c_p() {}
+
+  virtual void SetUp() override {
+    std::string platform_key = GetParam();
+    ASSERT_TRUE(test_platform::exists(platform_key));
+    platform_ = test_platform::get(platform_key);
+    system_ = test_system::instance();
+    system_->initialize();
+    system_->prepare_syfs(platform_);
+
+    log_set(stdout);
+
+    strcpy(tmpnull_gbs_, "tmpnull-XXXXXX.gbs");
+    close(mkstemps(tmpnull_gbs_, 4));
+
+    strcpy(invalid_gbs_, "invalid-XXXXXX.gbs");
+    close(mkstemps(invalid_gbs_, 4));
+
+    std::vector<uint8_t> gbs_hdr =
```

Comment:
should #include <vector> be included in the header?

Reply:
I think you mean
```c++
#include <vector>
```
I will add it.